extension/src/experiments/webview/messages.ts

```diff
@@ -248,6 +245,29 @@ export class WebviewMessages {
     }
   }
 
+  private async createAndSelectBranchFromExperiment(expId: string) {
+    const branchName = await commands.executeCommand(
+      RegisteredCliCommands.EXPERIMENT_VIEW_BRANCH,
+      { dvcRoot: this.dvcRoot, id: expId }
+    )
+
+    if (!branchName) {
+      return
+    }
+
+    await this.update()
+
+    const selectedBranches = this.experiments.getSelectedBranches()
+    const branchesWithNewBranch = this.experiments
+      .getAvailableBranchesToSelect()
+      .filter(
+        branch => selectedBranches.includes(branch) || branch === branchName
+      )
+
+    this.experiments.setSelectedBranches(branchesWithNewBranch)
```

Comment:
We only update the table if the user runs the context menu action. If the user runs the command in the command palette or the tree, we don't update the table. Figured it only made sense to update the table if the user could see the table, but we could also always update if preferred.

Reply:
Consistent behaviour is better, what if the tree is open, etc.

Reply:
Updated the behavior to always update the table!